services/catalog/src/simcore_service_catalog/api/rest/_services.py

```diff
@@ -105,56 +108,67 @@ async def list_services(
         )
 
     # now get the executable or writable services
-    services_in_db = {
-        (s.key, s.version): s
-        for s in await services_repo.list_services(
-            gids=[group.gid for group in user_groups],
-            execute_access=True,
-            write_access=True,
-            combine_access_with_and=False,
-            product_name=x_simcore_products_name,
-        )
-    }
+    with _tracer.start_as_current_span("services_repo.list_services") as span:
```

Comment:
Q: what happens in the catalog when tracing is disabled. Will this statement raise an error?

Reply:
I think I finally understood your comment in the description. You are saying that this has no influence on the code. This will continue to work as before

Reply:
The opentelemetry people seem to have thought of the use-case we discussed, that if tracing is disabled/disconnected, all tracing operations in the python code should be a no-op.
I have disabled tracing everywhere on my local `osparc-simcore`, running `make up-devel`, by setting in the `.env-devel`:
```
CATALOG_TRACING=null
TRACING_OPENTELEMETRY_COLLECTOR_BATCH_SIZE=1
TRACING_OPENTELEMETRY_COLLECTOR_PORT=
TRACING_OPENTELEMETRY_COLLECTOR_ENDPOINT=
```

I have added custom code in the catalog (in `services/catalog/src/simcore_service_catalog/api/rest/_services.py`), approximately like:
```
from opentelemetry import trace
_tracer = trace.get_tracer(__name__)
with _tracer.start_as_current_span("services_repo.list_services") as span:
    span.set_attribute("osparc-something", "set some attribute")
    do_stuff()
```


As seen in the screenshot from my local vscode remote-attached debugger, the opentelemetry people in this case automatically inject a `no-op tracer`, which does nothing, and also doesnt throw or log errors. It is exactly what we need. There are also no errors in the logs.

![image](https://github.com/user-attachments/assets/68debc1a-05c8-4c7a-a084-b52d2b912695)

So I would say we are free to add custom python code settings spans, traces to simcore, and be confident that this code is interpreted as a no-op if telemetry is turned off. :--)